### PR TITLE
Fix sandbox hot reloading

### PR DESCRIPTION
**Pull request by igor-sirotin**
### What does the PR do

Sandbox hot reloading wasn't working. Initially there were invalid watched paths, but also there were other major bugs:
- Multiple windows were appearing
- Tons of QML warnings about accessing `null` properties
- After 1-2 reloads Sandbox crashed

What I did is:
1. Watch QML file changes instead of directory changes
2. Store QML engine on heap and destroy it for fully singletons reloading
3. Preserve window position and size between reloads
4. Remove explicit `CMAKE_OSX_ARCHITECTURES "x86_64"`

### Affected areas

Sandbox

### Show time

https://user-images.githubusercontent.com/25482501/236399253-259153f7-0b8b-4dd9-bcc2-fe2ec88e3af2.mov

### CODE REVIEW

File: ui/StatusQ/sandbox/sandboxapp.h [hunk 1]
```diff
@@ -19,7 +19,7 @@ public slots:
     void restartEngine();
 
 private:
-    QQmlApplicationEngine m_engine;
+    QQmlApplicationEngine* m_engine {};
```
Comment (caybro) [inline ui/StatusQ/sandbox/sandboxapp.h:22]: `std::unique_ptr` for even more coolness :)

**Review by caybro — APPROVED** [review]
Nice... I had no idea it supported that feature :)

File: ui/StatusQ/sandbox/sandboxapp.h (hunk 1 above)
Comment (igor-sirotin) [reply ui/StatusQ/sandbox/sandboxapp.h:22]: oh, btw, I totally forgot to delete the engine on application close 🤦

Comment (borismelnik) [reply ui/StatusQ/sandbox/sandboxapp.h:22]: Which absolutely doesn't matter cos application closed? :D

**Review by borismelnik — APPROVED** [review]
LGTM 👍

**Comment by borismelnik** [comment]
> Nice... I had no idea it supported that feature :)

I thought we not using `SandboxApp` anymore :D

File: ui/StatusQ/sandbox/sandboxapp.cpp [hunk 2]
```diff
@@ -47,15 +47,15 @@ void SandboxApp::restartEngine()
         qDebug() << "Restarting QML engine";
 
     if (m_engine)
-        delete m_engine;
+        m_engine.release();
 
-    m_engine = new QQmlApplicationEngine();
+    m_engine = std::make_unique<QQmlApplicationEngine>();
```
Comment (caybro) [inline ui/StatusQ/sandbox/sandboxapp.cpp:52]: You could just use `m_engine.reset(new QQmlApplicationEngine)` w/o the `release()` call above

Comment (igor-sirotin) [reply ui/StatusQ/sandbox/sandboxapp.cpp:52]: indeed. That's what blind replacing does with us 😄

Comment (caybro) [reply ui/StatusQ/sandbox/sandboxapp.cpp:52]: And btw, `release()` does _not_ delete the raw pointer

Comment (caybro) [reply ui/StatusQ/sandbox/sandboxapp.cpp:52]: https://en.cppreference.com/w/cpp/memory/unique_ptr/release

**Pull request merged**